staking-bank/src/lib.rs

```diff
@@ -0,0 +1,510 @@
+fn _is_validator(_validator: AccountAddress) -> bool {
+    _validator == VALIDATOR_0 || _validator == VALIDATOR_1
+}
+
+#[cfg(feature = "sandbox")]
+/// Internal function that returns all validators.
+fn _addresses() -> Vec<AccountAddress> {
+    vec![VALIDATOR_0, VALIDATOR_1]
+}
+
+/// View function that returns validator's URL (as well as the inputted account address). The function throws an error if the address is not a validator.
+#[cfg(feature = "sandbox")]
+#[receive(
+    contract = "staking_bank",
+    name = "validators",
+    parameter = "AccountAddress",
+    return_value = "(AccountAddress,String)"
+)]
+fn validators<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    _host: &impl HasHost<State, StateApiType = S>,
+) -> ReceiveResult<(AccountAddress, String)> {
+    let id: AccountAddress = ctx.parameter_cursor().get()?;
+
+    match id {
+        VALIDATOR_0 => Ok((id, "https://validator.umb.network".to_string())),
+        VALIDATOR_1 => Ok((id, "https://validator2.umb.network".to_string())),
+        _ => bail!(CustomContractError::NotValidator.into()),
+    }
+}
+
+#[derive(Serial, Deserial)]
+pub struct State {}
+
+/// All smart contract errors.
+#[derive(Debug, PartialEq, Eq, Reject, Serial, SchemaType)]
+enum CustomContractError {
+    /// Failed to parse the parameter.
+    #[from(ParseError)]
+    ParseParams, // -1
+    /// Failed to log because the log is full.
+    LogFull, // -2
+    /// Failed to log because the log is malformed.
+    LogMalformed, // -3
+    /// Failed to invoke a contract.
+    InvokeContractError, // -4
+    /// Failed because the validators count is not consistent.
+    #[allow(dead_code)]
+    ValidatorsCountMisMatch, // -5
+    /// Failed because the address is not a validator.
+    #[allow(dead_code)]
+    NotValidator, // -6
+}
+
+/// Mapping errors related to logging to CustomContractError.
+impl From<LogError> for CustomContractError {
+    fn from(le: LogError) -> Self {
+        match le {
+            LogError::Full => Self::LogFull,
+            LogError::Malformed => Self::LogMalformed,
+        }
+    }
+}
+
+/// Mapping errors related to contract invocations to CustomContractError.
+impl<T> From<CallContractError<T>> for CustomContractError {
+    fn from(_cce: CallContractError<T>) -> Self {
+        Self::InvokeContractError
+    }
+}
+
+/// Init function that creates a new smart contract.
+#[cfg(any(feature = "production", feature = "development", feature = "sandbox"))]
+#[init(contract = "staking_bank")]
+fn init<S: HasStateApi>(
+    _ctx: &impl HasInitContext,
+    _state_builder: &mut StateBuilder<S>,
+) -> InitResult<State> {
+    let list = _addresses();
+
+    ensure_eq!(
+        list.len(),
+        NUMBER_OF_VALIDATORS as usize,
+        CustomContractError::ValidatorsCountMisMatch.into()
+    );
+
+    for validator in list {
+        ensure!(
+            _is_validator(validator),
+            CustomContractError::NotValidator.into()
+        );
+    }
```

Comment:
It was part of the solidity contracts. I see a very small value when you want to add new environments in the future. Because the tests will be run only on one environment, you might want to make sure that at least at the deployment of your new environment, you see if something is wrongly setup.

I decided to remove it because the contract size should be optimized on Concordium since it counts towards the execution cost and the above adds very minor value. 

@DZariusz